Ux library created

alexandercyu opened the pull request:
Adds:

- fonts
- colours
- backgrounds
- buttons
- username field
- password field
- visibility eye icon
- checkbox
- tooltips
- attention popup
- feedback popup
- network popup
- providers popup
- Main menu

mishaschwartz commented on templates/partials/button.html:
```diff
@@ -0,0 +1,90 @@
+{% set disabled = disabled_status|default('') %}
+
+{% set button_library = {
```
defining all these extra variables in this dictionary seems unnecessarily complicated. Can we just stick to "class" and "content" like we did in https://github.com/DACCS-Climate/marble-website/blob/main/templates/partials/button.html

alexandercyu replied:
Replaced most variables with "content", removed extra display conditions.

mishaschwartz commented on templates/site/js/login-styles-scripts.js:
```diff
@@ -0,0 +1,41 @@
+}
+
+
+window.onload = () => {
```
Is there a reason why you're doing this here instead of with the `$(document).ready` code above? (I know that they trigger at slightly different times but does that matter in this case?)

alexandercyu replied:
It was just developed before I got tooltips working and needed to import jquery.

mishaschwartz replied:
Ok makes sense. They should be merged now

alexandercyu replied:
Moved this functionality into jquery initialize function.

mishaschwartz commented on templates/partials/button.html:
```diff
@@ -0,0 +1,90 @@
+        },
+        "reg_saving_button": {
+            "class": "button-reg d-button-text",
+            "animated_spinner": '<i id="spinnerAnimated" class="fa-thin fa-circle-notch fa-spin visibility-spinner"></i>',
```
spinner does not appear without the `fa-solid` class as well

alexandercyu replied:
RIght.  I missed than the fa-thin version is for the pro version of fontawesome.

Class changed to fa-solid.

alexandercyu commented on templates/site/js/login-styles-scripts.js:
```diff
@@ -0,0 +1,41 @@
+
+window.onload = () => {
+    var userPassword = document.getElementById("userPassword");
+    var eyeIconContainer = userPassword.nextElementSibling;
```
It's because the div for the eye icon is in the username textfield as well, it's just hidden. So getting the eye icon div by id gets the hidden one and not the one I want.

I realize now I could have put a condition on the eye icon to include it dependent on the textfield type.

mishaschwartz replied:
The issue is that you have two elements with the id `eyeIconContainer` on the same page (ids *must* be unique)

alexandercyu replied:
Put the eye icon div inside a condition so it only displays in password field.

ShrutiK100 reviewed (changes requested):
Remove this button from design system
<img width="168" alt="Screenshot 2024-06-11 at 10 43 53 AM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/4ea6dd08-9033-4172-be03-a3b6cfdc4d24">

All text should be white, change title colour to white , body should be center aligned. Delete button has wrong elevation. Apply D-Elevation 3 to delete button,
<img width="567" alt="Screenshot 2024-06-11 at 10 49 09 AM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/b7dae888-4e32-41e9-8877-1d818a0a31f4">

alexandercyu commented:
> 
> All font are same as old design system, Is there is a reason why H1, H3 and H5 is defined separately in this design system?
>

All fonts are defined the  same as the old system.  No new fonts were defined.



> 
> New font - D-Button Text, D-ALT Button Text not defined and applied on the buttons. <img alt="Screenshot 2024-06-11 at 10 59 46 AM" width="784" src="https://private-user-images.githubusercontent.com/161752175/338639358-c2145bad-f10f-447b-8146-7afc5955331c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk0MzgsIm5iZiI6MTcxODExOTEzOCwicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjM5MzU4LWMyMTQ1YmFkLWYxMGYtNDQ3Yi04MTQ2LTdhZmM1OTU1MzMxYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxNTE4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MjljODlhZjRjOGUwMDI1ZDQwMzQ4YmYwYjRkZThiN2FkNzQ0OWZlMmYyYjU4YWRmMDI0NGI3ZjdiMmNjMDJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.49mzjknQJTVyC1RDjTXEUgrtZLCrIz3n5xp0FzFwBKk"> Define and apply the new fonts on the button text. <img alt="Screenshot 2024-06-11 at 11 00 13 AM" width="763" src="https://private-user-images.githubusercontent.com/161752175/338639402-cd38f1cb-58c6-4bdc-8da9-e70e4aa2bbc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk0MzgsIm5iZiI6MTcxODExOTEzOCwicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjM5NDAyLWNkMzhmMWNiLTU4YzYtNGJkYy04ZGE5LWU3MGU0YWEyYmJjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxNTE4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YWY2YjE5NmI2NjhmZTU1MzNiOTU4MTU4MjU0Mjc4MDMzYThjZGNkM2Q0ZDE2YmEwZTU2YTA5NTRkZTg2ZmExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CHTZKRf-TnnAOk-X_OOCKJaRS7p_W_U0mHYC_bxrXzw">
>

The font for the Regular button has already been defined.  Am I missing something for it?
![Screenshot 2024-06-11 at 11 58 09 AM (2)2](https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/124736788/7f6b162f-9513-40a0-bfba-edcbbcddf409)

alexandercyu commented:
> Remove this button from design system <img alt="Screenshot 2024-06-11 at 10 43 53 AM" width="168" src="https://private-user-images.githubusercontent.com/161752175/338633529-4ea6dd08-9033-4172-be03-a3b6cfdc4d24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTg1OTUsIm5iZiI6MTcxODExODI5NSwicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjMzNTI5LTRlYTZkZDA4LTkwMzMtNDE3Mi1iZTAzLWEzYjZjZmRjNGQyNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxNTA0NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWE3MTM0ZTZmYWUzZDg1YTFmYWYyYjQyMTI3YjA1MTQ3YWY3NTE3ZDE0ODUyZTBmOTlmNDE3YWQwYWM4ZjUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0RjD-PpU4yHj4sZ4pWEjHW16xXF-FAOlfZmb-YWbRo4">
> 

Button removed as per changes to design.

ShrutiK100 commented:
> D Button Text is alright! that seems to be applied 
please check:
D-ALT Button Text is not applied on provider's buttons.

alexandercyu commented:
> Check box doesn't look the same as in the design <img alt="Screenshot 2024-06-11 at 10 59 04 AM" width="56" src="https://private-user-images.githubusercontent.com/161752175/338638196-4b5e2034-5dc7-4fa2-8b09-0125f48f6092.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMzQ5OTcsIm5iZiI6MTcxODEzNDY5NywicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjM4MTk2LTRiNWUyMDM0LTVkYzctNGZhMi04YjA5LTAxMjVmNDhmNjA5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxOTM4MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNjlmNzNmODJhNGM5YTcwNmRmYjNmNWNiNmRmOGVmYWI3OTM0YmJmM2QzOTRmMTI3YTJlNGJiYzcyYTc5MDg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.C1kuvhy_rogyzP0fRA_Scaj3uCZxvcgPOq4ADi9Exio"> <img alt="Screenshot 2024-06-11 at 10 56 34 AM" width="208" src="https://private-user-images.githubusercontent.com/161752175/338638152-c1cdcd25-90ab-423f-8a52-b1b83102c351.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMzQ5OTcsIm5iZiI6MTcxODEzNDY5NywicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjM4MTUyLWMxY2RjZDI1LTkwYWItNDIzZi04YTUyLWIxYjgzMTAyYzM1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxOTM4MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNWIwOWQ0YzBmZWU1ZDY0YTJmMTkwN2FiODgwZjc4ZTZmMWU3OTA1MGQ5YzExNmRjYzA0N2Q2ZTRiYjkyM2MwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2gNq-CBg1_OcZWSlmVLsY1wS6JyVRjccN0Ze5e_tWBM">
> 


Adjusted look of checkmark

alexandercyu commented on templates/layouts/main.html:
```diff
@@ -0,0 +1,37 @@
+    <meta charset="UTF-8">
+    <title>{{ page_title }}</title>
+</head>
+<body class="d-flex">
```
This was copied straight from the node landing page without changes.

Removed this class.

alexandercyu commented on templates/site/index.html:
```diff
@@ -0,0 +1,146 @@
+    </header>
+
+    <div class="d-flex flex-column justify-content-evenly">
+        <h1>Marble @ {{ node_name }}</h1>
```
Removed Marble title.

alexandercyu commented on templates/site/index.html:
```diff
@@ -0,0 +1,146 @@
+        <br/>
+
+        {% with button_type = "med_action_alt_button"%}
+            {% set button_image = "images/buttons/img_swapcard_1.png" %}
```
Path fixed

alexandercyu commented on templates/site/index.html:
```diff
@@ -0,0 +1,146 @@
+
+        {% with button_type = "med_action_alt_button"%}
+            {% set disabled_status = true %}
+            {% set button_image = "images/buttons/img_swapcard_1.png" %}
```
Path fixed

alexandercyu commented on templates/partials/popup-attention.html:
```diff
@@ -0,0 +1,14 @@
+<div class="popup-attention-container">
+    <div class="popup-attention-content-container">
+        <h4>{{ popup_attention_title }}</h4>
+        <div class="popup-attention-logo margin-popup-content"></div>
```
Logo fixed

alexandercyu commented on templates/partials/checkbox.html:
```diff
@@ -0,0 +1,19 @@
+{% set input_checkbox_id = checkbox_id %}
```
Removed variables for now.  I think this for a future use case where multiple checkboxes would be needed and their ids would need to be assigned dynamically.  But that can be created at that time.

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+{% set input_textbox_type = textbox_type %}
```
Changed to no longer rename variable.

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+{% set input_textbox_type = textbox_type %}
+
+{% if textbox_type == "password_input_text" %}
```
Changed to no longer rename variable.

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+            "input_required": "required ",
+            "eye_icon_visible": "",
+            "more_info_title": "",
+            "more_info_text": "Password must be 12 characters long."
```
Changed text to a placeholder.

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+            "class": "input-textbox",
+            "input_type": "type = text",
+            "eye_icon_visible": "visibility-none-password-eye",
+            "more_info_title": "",
```
It was the default setting of the tooltip.  I thought it had to be kept blank in order for the title to not be shown.

alexandercyu replied:
Removed this entry

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+        "reg_input_text": {
+            "textbox_label": "Email or Username",
+            "textbox_name": "username",
+            "class": "input-textbox",
```
Removed from variable list.

alexandercyu commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,54 @@
+            "textbox_label": "Password",
+            "textbox_name": "userpassword",
+            "class": "input-textbox",
+            "input_type": "type = password",
```
There was an issue with the quotes.  If the value was just "password" or "text" it would render with extra quotes and look something like '"password"', and it wouldn't work properly.

alexandercyu commented on templates/site/js/login-styles-scripts.js:
```diff
@@ -0,0 +1,41 @@
+    }
+}
+
+function toggleEyeIcon(element) {
```
There's no case where they are called separately, but shouldn't they be in different functions so testing and troubleshooting is easier?

mishaschwartz replied:
ok sure

alexandercyu commented on templates/site/js/toggle-password.js:
This was the first version. Then I put everything together in one file.

This file is deleted.

alexandercyu commented on templates/partials/button.html:
```diff
@@ -0,0 +1,90 @@
+                type="button" onclick="{{ button_library[button_type]['show_spinner_function'] }}" >
+
+            <!-- Used for Saving button, Saved button -->
+            {% if button_library[button_type]['animated_spinner'] is defined %}
```
Removed the "is defined" portion

alexandercyu commented on templates/partials/button.html:
```diff
@@ -0,0 +1,90 @@
+{% set disabled = disabled_status|default('') %}
```
Changed so not renaming variable.

alexandercyu commented:
> All text should be white, change title colour to white , body should be center aligned. Delete button has wrong elevation. Apply D-Elevation 3 to all delete buttons. <img alt="Screenshot 2024-06-11 at 10 49 09 AM" width="567" src="https://private-user-images.githubusercontent.com/161752175/338634422-b7dae888-4e32-41e9-8877-1d818a0a31f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDQwNTAsIm5iZiI6MTcxODY0Mzc1MCwicGF0aCI6Ii8xNjE3NTIxNzUvMzM4NjM0NDIyLWI3ZGFlODg4LTRlMzItNDFlOS04ODc3LTFkODE4YTBhMzFmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxN1QxNzAyMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMDY4ZDJjMTIzMmU5OTNmMTMwNDA3ZjlhMTRiNzFjYzYxZmEyN2VmMjM3YzA1YmU1MGYxNDBhMmU5MzE3ZDU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Q8MZntK8ZFuiQ-SCTuJGeg_cEujBkhdFPOWuqntuUXo">
> 

Title changed to white.

Text centre aligned.

mishaschwartz commented on templates/partials/textbox.html:
```diff
@@ -0,0 +1,51 @@
+{% set textbox_type = textbox_type %}
```
this does nothing

mishaschwartz commented on templates/layouts/main.html:
```diff
@@ -0,0 +1,37 @@
+
+
+    <!--Custom scripts -->
+    <script src="js/login-styles-scripts.js"></script>
```
Does this need to be loaded on every page or can we load this as needed

alexandercyu replied:
It could be loaded as needed.  I think I'll need to look at how the lightbox/modal will be loaded to see where the script will be called from.

mishaschwartz commented on templates/layouts/main.html:
```diff
@@ -0,0 +1,37 @@
+    <script src="js/login-styles-scripts.js"></script>
+
+    <link href="css/node-login.css" rel="stylesheet">
+    <link rel="icon" type="image/x-icon" href="images/favicon.ico">
```
this image doesn't exist

alexandercyu replied:
Uploaded the favicon, which right now is the Marble logo.

mishaschwartz reviewed (changes requested):
Ok a few small things and otherwise the code itself looks good. 

The only thing is that there's a lot of the content here that should be set externally and/or through a configuration file. I don't know if this is the right place to make that change though. I'm happy to merge this in with the current content as placeholders and then we can work on making it dynamic later on.

alexandercyu commented:
As per yesterday's meeting:
- Added space for text in textfield
- Increased space for eye icon in textfield
- Added small amount of padding for feedback entry logo so there is space on hover
- Changed colour and hover colour of button with cut-off corner to primary

alexandercyu commented:
> The logo quality has deteriorated.
> <img alt="Screenshot 2024-06-25 at 1 12 20 PM" width="653" src="https://private-user-images.githubusercontent.com/161752175/342911450-671452a4-cd02-4c14-b833-a837359eb175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNDY3NzksIm5iZiI6MTcxOTM0NjQ3OSwicGF0aCI6Ii8xNjE3NTIxNzUvMzQyOTExNDUwLTY3MTQ1MmE0LWNkMDItNGMxNC1iODMzLWE4MzczNTllYjE3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNVQyMDE0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjZlZDBiNmVmOTgxYjcyYTdiNjdlNGMyNTMzOTU4MmFiYTQ1ZmI3ODk3N2VkNDU1YzBjMDQyMjE3MDI4NDZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ApM5JSL851gWnz74CbamgLacIMkK65K_Se476QuTtjk">

I got these from figma.  How has the logo deteriorated?

ShrutiK100 commented:
> > The logo quality has deteriorated.
> > <img alt="Screenshot 2024-06-25 at 1 12 20 PM" width="653" src="https://private-user-images.githubusercontent.com/161752175/342911450-671452a4-cd02-4c14-b833-a837359eb175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNDY3NzksIm5iZiI6MTcxOTM0NjQ3OSwicGF0aCI6Ii8xNjE3NTIxNzUvMzQyOTExNDUwLTY3MTQ1MmE0LWNkMDItNGMxNC1iODMzLWE4MzczNTllYjE3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNVQyMDE0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjZlZDBiNmVmOTgxYjcyYTdiNjdlNGMyNTMzOTU4MmFiYTQ1ZmI3ODk3N2VkNDU1YzBjMDQyMjE3MDI4NDZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ApM5JSL851gWnz74CbamgLacIMkK65K_Se476QuTtjk">
> 
> I got these from figma. How has the logo deteriorated?
> 
> Additionally, none of the logos have changed. Only the Github one was replaced with the black version.

Which format were the logos downloaded as? svg or png?

alexandercyu commented:
> > > The logo quality has deteriorated.
> > > <img alt="Screenshot 2024-06-25 at 1 12 20 PM" width="653" src="https://private-user-images.githubusercontent.com/161752175/342911450-671452a4-cd02-4c14-b833-a837359eb175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNDY3NzksIm5iZiI6MTcxOTM0NjQ3OSwicGF0aCI6Ii8xNjE3NTIxNzUvMzQyOTExNDUwLTY3MTQ1MmE0LWNkMDItNGMxNC1iODMzLWE4MzczNTllYjE3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNVQyMDE0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjZlZDBiNmVmOTgxYjcyYTdiNjdlNGMyNTMzOTU4MmFiYTQ1ZmI3ODk3N2VkNDU1YzBjMDQyMjE3MDI4NDZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ApM5JSL851gWnz74CbamgLacIMkK65K_Se476QuTtjk">
> > 
> > 
> > I got these from figma. How has the logo deteriorated?
> > Additionally, none of the logos have changed. Only the Github one was replaced with the black version.
> 
> Which format were the logos downloaded as? svg or png?


![Screenshot 2024-06-26 at 11 32 08 AM (3)](https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/124736788/c8bb1076-eb2c-49b9-9a90-68e94eab25d1)


The dropdown only shows png, jpg or pdf formats.

alexandercyu commented:
> > > > The logo quality has deteriorated.
> > > > <img alt="Screenshot 2024-06-25 at 1 12 20 PM" width="653" src="https://private-user-images.githubusercontent.com/161752175/342911450-671452a4-cd02-4c14-b833-a837359eb175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNDY3NzksIm5iZiI6MTcxOTM0NjQ3OSwicGF0aCI6Ii8xNjE3NTIxNzUvMzQyOTExNDUwLTY3MTQ1MmE0LWNkMDItNGMxNC1iODMzLWE4MzczNTllYjE3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNVQyMDE0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjZlZDBiNmVmOTgxYjcyYTdiNjdlNGMyNTMzOTU4MmFiYTQ1ZmI3ODk3N2VkNDU1YzBjMDQyMjE3MDI4NDZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ApM5JSL851gWnz74CbamgLacIMkK65K_Se476QuTtjk">
> > > 
> > > 
> > > I got these from figma. How has the logo deteriorated?
> > > Additionally, none of the logos have changed. Only the Github one was replaced with the black version.
> > 
> > 
> > Which format were the logos downloaded as? svg or png?
> 
> ![Screenshot 2024-06-26 at 11 32 08 AM (3)](https://private-user-images.githubusercontent.com/124736788/343227784-c8bb1076-eb2c-49b9-9a90-68e94eab25d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MTYzNjEsIm5iZiI6MTcxOTQxNjA2MSwicGF0aCI6Ii8xMjQ3MzY3ODgvMzQzMjI3Nzg0LWM4YmIxMDc2LWViMmMtNDliOS05YTkwLTY4ZTk0ZWFiMjVkMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxNTM0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMjc0ZTlhZDY2MjllMGJkMzM4ZjM4YjE3YmZlNjVhYTk5MjQzZDllY2Y3MmQ5NGNjNjg1YjU1MTM4MzdjODNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nKge__0PWWBHJZzFHu47A7Y3O2FS1ZvK_wtBC79_FfA)
> 
> The dropdown only shows png, jpg or pdf formats.


![Screenshot 2024-06-26 at 11 57 45 AM (3)](https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/124736788/1d473868-344f-4369-953d-e0f9628d562a)


I got the svg versions of these logos now.

ShrutiK100 commented:
The logo on hover looks different than design
<img width="179" alt="Screenshot 2024-06-26 at 2 02 40 PM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/786f4838-4df9-445e-9e6b-2631b066d571">
<img width="386" alt="Screenshot 2024-06-26 at 2 07 19 PM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/a21e4de5-4e4b-4176-ac50-7b0a1537a12b">

ShrutiK100 commented:
Save changes button have 4 states- default, hover, saving and saved, Only default state the button colour is white, all other states should be primary colour. Is there a reason why Saved button has been created seperately?
<img width="205" alt="Screenshot 2024-06-26 at 2 13 02 PM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/abc594ca-9eda-4243-b96a-b6ab7fc583f9">

alexandercyu commented:
> Save changes button have 4 states- default, hover, saving and saved, Only default state the button colour is white, all other states should be primary colour. Is there a reason why Saved button has been created seperately? <img alt="Screenshot 2024-06-26 at 2 13 02 PM" width="205" src="https://private-user-images.githubusercontent.com/161752175/343284401-abc594ca-9eda-4243-b96a-b6ab7fc583f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MjYyMjIsIm5iZiI6MTcxOTQyNTkyMiwicGF0aCI6Ii8xNjE3NTIxNzUvMzQzMjg0NDAxLWFiYzU5NGNhLTllZGEtNDI0My1iOTZhLWI2YWI3ZmM1ODNmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxODE4NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZmM1ZjhiODIwZjg1MGUwMDBkNDEzZWM0Y2VlMDk0ODJmNGFjN2MzMjVjYWU2MTZiNDJjMGVmY2M3NGRjOTdjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.L6U28WPCPUVx3QtHWp8GNgJsxJ3GKTT8nNUFW8y6P4I">

To show the design of the button.

ShrutiK100 commented:
> > Save changes button have 4 states- default, hover, saving and saved, Only default state the button colour is white, all other states should be primary colour. Is there a reason why Saved button has been created seperately? <img alt="Screenshot 2024-06-26 at 2 13 02 PM" width="205" src="https://private-user-images.githubusercontent.com/161752175/343284401-abc594ca-9eda-4243-b96a-b6ab7fc583f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MjYyMjIsIm5iZiI6MTcxOTQyNTkyMiwicGF0aCI6Ii8xNjE3NTIxNzUvMzQzMjg0NDAxLWFiYzU5NGNhLTllZGEtNDI0My1iOTZhLWI2YWI3ZmM1ODNmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxODE4NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZmM1ZjhiODIwZjg1MGUwMDBkNDEzZWM0Y2VlMDk0ODJmNGFjN2MzMjVjYWU2MTZiNDJjMGVmY2M3NGRjOTdjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.L6U28WPCPUVx3QtHWp8GNgJsxJ3GKTT8nNUFW8y6P4I">
> 
> To show the design of the button.

I see. Please change the colour of Saved button to primary.

alexandercyu commented:
> The Github logo on hover looks different than design <img alt="Screenshot 2024-06-26 at 2 02 40 PM" width="179" src="https://private-user-images.githubusercontent.com/161752175/343281412-786f4838-4df9-445e-9e6b-2631b066d571.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MjYyMjIsIm5iZiI6MTcxOTQyNTkyMiwicGF0aCI6Ii8xNjE3NTIxNzUvMzQzMjgxNDEyLTc4NmY0ODM4LTRkZjktNDQ1ZS05ZTZiLTI2MzFiMDY2ZDU3MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxODE4NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNDZmMzI4NWUxZWU0ZjA3ODQwZTBhNGU2OTUwYmQzN2RmOGZmZmMyNmYxZDc3ZjgzNTg4NTViMTU5N2FhYTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._LL9L9QxEed_Pzgfei3i8ucLFQXk1GXZTLUrVTe8dPA"> <img alt="Screenshot 2024-06-26 at 2 07 19 PM" width="386" src="https://private-user-images.githubusercontent.com/161752175/343281587-a21e4de5-4e4b-4176-ac50-7b0a1537a12b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MjYyMjIsIm5iZiI6MTcxOTQyNTkyMiwicGF0aCI6Ii8xNjE3NTIxNzUvMzQzMjgxNTg3LWEyMWU0ZGU1LTRlNGItNDE3Ni1hYzUwLTdiMGExNTM3YTEyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxODE4NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01N2FlNDQyNjJhN2M2NjA5ZTM1MjFjOWY3ZGI2ZjcxYWNiZDlhZjYzNzJkODg3MmIyMGE1MTM2ZDY2NWU1OGQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JE3hfPy7LMleJ3b9m2vFX8CniT0ZW0eqxgVOZVDhevw">

Changed github logo for hover state.

ShrutiK100 commented:
<img width="1007" alt="Screenshot 2024-07-02 at 11 03 52 AM" src="https://github.com/DACCS-Climate/Marble-node-Magpie-ui/assets/161752175/ce1dcceb-4f48-43a0-b631-c8b5f9d4e254">
 Please make these final changes.

alexandercyu commented:
> <img alt="Screenshot 2024-07-02 at 11 03 52 AM" width="1007" src="https://private-user-images.githubusercontent.com/161752175/345087449-ce1dcceb-4f48-43a0-b631-c8b5f9d4e254.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MzMwMDMsIm5iZiI6MTcxOTkzMjcwMywicGF0aCI6Ii8xNjE3NTIxNzUvMzQ1MDg3NDQ5LWNlMWRjY2ViLTRmNDgtNDNhMC1iNjMxLWM4YjVmOWQ0ZTI1NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQxNTA1MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNTA0YTkxZmE1YzlhZWFmNTZkMjc1NDc5NjE3YmRlYTBiMjUxZDBiYTM1MTdlNzQ1NTQxM2U5MmU2ZWZmNTgxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6Chidk2aP2ygVsV9TQ8jppqvHti5Eaa17wMKYHCv2fQ"> Please make these final changes.

Added the padding.

ShrutiK100 reviewed (approved):
Looks good!
